rest/main/main.go

```diff
@@ -105,8 +105,6 @@ func main() {
 		Certificates:             prepareServerCertificate(),
 		ClientCAs:                prepareCACertificates(),
 		MinVersion:               tls.VersionTLS12,
-		PreferServerCipherSuites: true,
-		CipherSuites:             getPreferredCipherSuites(),
```

Comment:
Did you see any issue using TLS1.3 clients? Golang uses this **CipherSuites** setting only for TLS1.0-1.2 communications. It is ignored when TLS1.3 is selected -- server automatically picks the best cipher. Current configuration forces stronger ciphers when TLS1.2 is selected.
If you want to force TLS1.3 only, please change **MinVersion** configuration to `tls.VersionTLS13` at line 107.

Reply:
If `CipherSuites` is not specified, then a safe default cipher suites list is used. I think using default list is better? Cipher suites from this `getPreferredCipherSuites` may not be safe over time.

Reply:
The ciphers returned by `getPreferredCipherSuites()` are recommended by security team; but I agree it was almost 3 years back. I am okay to remove it and let golang library decide the best values (assuming we upgrade the golang version to latest).
But, my point was that it should not have affected TLS1.3 communication. Golang TLS library ignores the CipherSuites values when the client selects TLS1.3 ([API doc](https://cs.opensource.google/go/go/+/refs/tags/go1.15.5:src/crypto/tls/common.go;l=611)). So, the TLS1.3 communication should have worked without these changes too. Did you see any failures with the old code?

Reply:
I agree that it should not have impacted TLS1.3 communication. The requirement is must support TLS 1.3 by default, if not possible, fall back to TLS 1.2. We remove `getPreferredCipherSuites` ciphers to ensure default ciphers remain secure in case TLS clients/ servers negotiates TLS 1.2.

Reply:
I understand the requirements, but the changes have nothing to do with it. REST server already supports TLS1.2 and TLS1.3 as expected.

Reply:
I agree server already supports TLS 1.2 and TLS 1.3. I'm confused why "the changes have nothing to do with it".
For TLS 1.3, server automatically picks the best cipher.
For TLS 1.2, the changes in this PR (and upgrade Golang to version 1.15) will pick default cipher suites when TLS 1.2 is selected, correct? I'm wondering why not use same practice for both TLS 1.2 and TLS 1.3?

Reply:
hi @maipbui, I am okay to remove the `CipherSuites` preference. But we need to keep `PreferServerCipherSuites` for now (until we upgrade to go1.17 or later). Without it the server picks client's CipherSuite order and can end up using a weak cipher today.

Reply:
I see your point, will update the PR.